command/server.go

```diff
@@ -351,6 +353,13 @@ func (c *ServerCommand) Run(args []string) int {
 		}
 	}
 
+	// Make sure we close all listeners from this point on
+	defer func() {
```

Comment:
You can see below in the diff where we already had this code copied twice...this just moves it into a common function.